src/Components/Filter/Filter.js

```diff
@@ -0,0 +1,208 @@
+  let data = [...productList];
+  if (state.filterBy === "1STAR_AND_ABOVE") {
+    data = [...productList].filter((product) => product.rate >= 1);
+  }
+  if (state.filterBy === "2STAR_AND_ABOVE") {
+    data = [...productList].filter((product) => product.rate >= 2);
+  }
+  if (state.filterBy === "3STAR_AND_ABOVE") {
+    data = [...productList].filter((product) => product.rate >= 3);
+  }
+  if (state.filterBy === "4STAR_AND_ABOVE") {
+    data = [...productList].filter((product) => product.rate >= 4);
+  }
+  return data;
+};
+
+export const getFilteredData = (state, productList) => {
+  let data = [...productList];
+  if (state.FilterData.filterByCategories.length !== 0) {
+    data = data.filter((product) =>
+      state.FilterData.filterByCategories.includes(product.categoryName)
+    );
+  }
+
+  return data;
+};
+
+export const getFilterByPrice = (state, productList) => {
+  let data = [...productList];
+  data = data.filter((product) => product.price >= state.getByPrice);
+  return data;
+};
+
+export function Filter() {
+  const { state, dispatch } = useProduct();
+
+  return (
+    <div className="filter">
+      <div className="filter-heading">
+        <div className="semibold-font-weight">FILTERS</div>
+        <button className="clear-all-btn bold-font-weight" onClick={() =>
+          {
+            dispatch({
+              type:"CLEAR"
+            })
+          }
+        }>CLEAR</button>
+      </div>
+      <div className="price-section">
+        <div className="price semibold-font-weight price-title">Price</div>
+        <div className="range">
+          <span>100</span>
+          <span>500</span>
+          <span>1000</span>
+        </div>
+        <input
+          type="range"
+          min="100"
+          max="1000"
+          onChange={(e) => {
+            dispatch({
+              type: "FILTER_BY_PRICE",
+              payload: e.target.value,
+            });
+          }}
+          className="slider"
+        />
+      </div>
+      <div className="filter-category">
+        <div className="semibold-font-weight category-title">Category</div>
+        {categories.map((category) => {
+          return (
+            <div className="inline-category-item">
+              <input
+                type="checkbox"
+                checked={state.FilterData.filterByCategories.includes(
+                  category.categoryName
+                )}
+                className="filter-input"
+                onChange={() =>
+                  dispatch({
+                    type: "FILTER_BY_CATEGORY",
+                    payload: category.categoryName,
+                  })
+                }
+              />
+              <label>{category.categoryName}</label>
+            </div>
+          );
+        })}
+      </div>
+
+      <div className="rating-section">
+        <div className="rating-title semibold-font-weight">Rating</div>
+        <div className="inline-category-item">
+          <input
```

Comment:
Add input inside label. it will be a good user experience. It is a bad experience to let users click on that small checkbox or small radio button only. if you put input inside the label, even if you click on label it will checked

Reply:
Done